src/fhirtypes/common.ts

```diff
@@ -66,9 +66,15 @@ export function setPropertyOnInstance(
         if (current[key] == null) current[key] = [];
         sliceName = getSliceName(pathPart);
         if (sliceName) {
+          if (typeof fixedValue !== 'object') {
+            // When a fixedValue is a primitive but also a slice, we convert to an object so that
+            // the sliceName field can be tracked on the object. The _primitive field marks the object
+            // to later be converted back to a primitive by replaceField in cleanInstance
+            fixedValue = { fixedValue, _primitive: true };
+          }
           const sliceIndices: number[] = [];
           // Find the indices where slices are placed
-          current[key].forEach((el: any, i: number) => {
+          current[pathPart.base].forEach((el: any, i: number) => {
```

Comment:
I think I'm not quite understanding how this works when we have a primitive. Since the key will be `_${pathPart.base}` , then how is it that `current[pathPart.base]` even exists at this point?

Reply:
Hm yes, this is part of a larger overall problem that I now see. You can access the child of a primitive even if the primitive itself doesn't exist. For example, the following FSH snippet:
```
* name[0].prefix[0].id = "oops"
```
does not give you any errors, even if `name[0].prefix[0]` does not yet exist and you end up with output like:
```
"name": [
    {
      "_prefix": [
        {
          "id": "oops"
        }
      ]
    }
  ]
```
which clearly shouldn't be allowed. So I will add logic to check that `current[pathPart.base]` does indeed exist, or else we shouldn't be adding it's so called "child" to the instance.

Reply:
Actually I am not sure. Should that not be allowed? You can do:
```
* name[0].prefix[0] = "Dr."
```
And the name array will be populated. It isn't like you need to make the name array before setting its children. So I don't think you should have to make the primitive before making its children. I don't know what the expected behavior would be because primitives having children that are tracked as sibling properties with an `_` appended is already a bit weird. What do you think?

Reply:
Hm it looks like you actually can have a primitive extension (so something like `_prefix`) without having a fixed value for it:
>Note: If the primitive has an id attribute or extension, but no value, only the property with the _ is rendered.

(from: https://www.hl7.org/fhir/json.html#primitive)

Given that , maybe this isn't an issue and can stay how it is?

Reply:
Ok well that's weird, but yeah I think that answers that question then. Thanks!